Makefile

```diff
@@ -15,17 +16,19 @@ CFLAGS_ALL=$(CFLAGS_SAML) -Wall -Werror -std=c99 $(XMLSEC1_CFLAGS)
 LIBFLAG=-shared
 LDFLAGS=-g -O2
 XMLSEC1_STATIC_LIBS=xmlsec1-$(XMLSEC_VER)/./src/openssl/.libs/libxmlsec1-openssl.a xmlsec1-$(XMLSEC_VER)/./src/.libs/libxmlsec1.a
-XMLSEC1_LDFLAGS=-Wl,--whole-archive $(XMLSEC1_STATIC_LIBS) -Wl,--no-whole-archive -lxml2 -lssl -lcrypto -ldl -lxslt
+XMLSEC1_LDFLAGS=-Wl,--whole-archive $(XMLSEC1_STATIC_LIBS) -Wl,--no-whole-archive -L$(OPENSSL_DIR)/lib/ -lxml2 -lssl -lcrypto -ldl -lxslt -Wl,-rpath $(OPENSSL_DIR)/lib
```

Comment:
We should rely on the OPENSSL_XX variable passed by luarocks. See https://github.com/brunoos/luasec/blob/d9215ee00f6694a228daad50ee85827a4cd13583/luasec-1.2.0-1.rockspec#L52 and https://github.com/apache/apisix/blob/b48d02051a87abcf2ab6ea7c8cc2286db40ef6e6/Makefile#L155

Reply:
@kingluo done